syntax = "proto2";

package rv.protos;

import "rastervision/protos/machine_learning.proto";
import "rastervision/protos/scene.proto";
import "rastervision/protos/make_training_chips.proto";
import "rastervision/protos/train.proto";
import "rastervision/protos/predict.proto";
import "rastervision/protos/eval.proto";
import "rastervision/protos/raster_transformer.proto";

message ChainWorkflowConfig {
    repeated Scene train_scenes = 1;

    // The test_scenes are used in the eval command for measuring the
    // quality of models, and are used to generate validation chips to use
    // during training.
    repeated Scene test_scenes = 18;

    required MachineLearning machine_learning = 3;
    required RasterTransformer raster_transformer = 5;
    required int32 chip_size = 6;
    required bool debug = 7;

    // Map of parameters to values to use when doing parameter substitution on
    // generated URIs. One of the keys must be 'rv_root' which is used when
    // generating the URI for all the output generated by running RV.
    map<string, string> local_uri_map = 8;
    map<string, string> remote_uri_map = 9;

    // The names of the raw dataset, (processed) dataset, model, prediction,
    // and eval. These are used to generate URIs.
    required string raw_dataset_key = 19;
    required string dataset_key = 10;
    required string model_key = 11;
    required string prediction_key = 12;
    required string eval_key = 13;

    required MakeTrainingChipsConfig.Options make_training_chips_options = 14;
    required TrainConfig.Options train_options = 15;
    required PredictConfig.Options predict_options = 16;
    required EvalConfig.Options eval_options = 17;
}
